config/config-network.yaml

```diff
@@ -98,8 +98,8 @@ data:
     # entirely from the template. When choosing a new value be thoughtful
     # of the potential for conflicts - for example, when users choose to use
     # characters such as `-` in their service, or namespace, names.
-    #
-    domainTemplate: "{{.Name}}.{{.Namespace}}.{{.Domain}}"
+    # SubDomains can be passed via label "serving.knative.dev/subDomain", if set will be used else will fall back to namespace 
+    domainTemplate: "{{if eq (len .SubDomain) 0}}{{.Name}}.{{.Namespace}}.{{.Domain}}{{ else }}{{.Name}}.{{.SubDomain}}.{{.Domain}}{{ end }}"
```

Comment:
I wonder if we should add the comment but leave the domain template unchanged so users can choose how best to use the subdomain (in addition to namespace, before or after namespace etc.)